src/Common/CounterInFile.h

```diff
@@ -112,11 +113,13 @@ class CounterInFile
         }
         catch (...)
         {
-            close(fd);
+            int err = close(fd);
+            chassert(!err || errno == EINTR);
```

Comment:
Consider adding an inline function somewhere. This function will simply call `close`, check the error with `chassert`, and contain a comment about retrying EINTR on Linux. But it's not strictly necessary.

Reply:
I thought about this, but did not like the idea of having such a function (mostly because of the naming issue and that fact that developers will need to know that there is some special function), anyway there a simple style check to disallow close() without check.